extensions/wikia/AdEngine3/AdEngine3ApiController.class.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+class AdEngine3ApiController extends WikiaController
+{
+	const DEFAULT_TEMPLATE_ENGINE = WikiaResponse::TEMPLATE_ENGINE_MUSTACHE;
+
+	public function getRecCode()
+	{
+		global $wgUser;
+
+		$this->response->setContentType('text/javascript');
+		$this->response->setCachePolicy(WikiaResponse::CACHE_PUBLIC);
+		$this->response->setCacheValidity(WikiaResponse::CACHE_LONG);
+
+		if ($wgUser->isAnon()) {
+			$resourceLoader = null;
+			$type = $this->request->getVal('type', 'rec');
+
+			switch ($type) {
+				case 'bt':
+					$resourceLoader = new ResourceLoaderAdEngine3BTCode();
+					break;
+
+				case 'hmd':
+					$resourceLoader = new ResourceLoaderAdEngine3HMDCode();
+					break;
+			}
+
+			if ($resourceLoader) {
+				$resourceLoaderContext = new ResourceLoaderContext(new ResourceLoader(), new FauxRequest());
+				$this->response->setBody($resourceLoader->getScript($resourceLoaderContext));
```

Comment:
`$resourceLoader` can be null. Should we respond some 403 for that case?

Reply:
Nah, I think that empty response is better and safer, because in case of cache happen it will still return 200 response code, but without any code, which can be executed.

Reply:
+ Made some cleanup

Reply:
Imo `403` for not defined type is fine, but let's leave it